src/chat/flagsReducers.js

```diff
@@ -0,0 +1,58 @@
+import {
```

Comment:
We both need to invalidate referentially the array itself, and the entry.
In the sense of:

```js
const newArray = oldArray.slice();
newArray !== oldArray
```

That is a perfectly orthodox approach. Don't be concerned about perf; slicing 1000 items takes about 0.05msecs so kinda instant.